### PR TITLE
LPD-15716 Add to the SEO tab of each Page the option to exclude Child Pages

### PR DESCRIPTION
**Pull request by lfbesada**
https://liferay.atlassian.net/browse/LPD-15716

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/layout/layout-seo-web/src/main/resources/META-INF/resources/layout/screen/navigation/entries/seo.jsp [hunk 1]
```diff
@@ -210,6 +210,12 @@ UnicodeProperties layoutTypeSettingsUnicodeProperties = selLayout.getTypeSetting
 							name="TypeSettingsProperties--sitemap-changefreq--"
 							options="<%= layoutsSEODisplayContext.getSitemapChangeFrequencySelectOptions() %>"
 						/>
+
+						<c:if test="<%= layoutsSEODisplayContext.showIncludeChildPagesInSitemap() %>">
+							<aui:input checked="<%= layoutsSEODisplayContext.isIncludeChildPagesInSitemap() %>" label="include-child-page-urls-in-the-xml-sitemap" labelCssClass="font-weight-normal" name="TypeSettingsProperties--sitemap-include-child-pages--" type="checkbox" wrapperCssClass="mb-2" />
```
Comment (ealonso) [inline modules/apps/layout/layout-seo-web/src/main/resources/META-INF/resources/layout/screen/navigation/entries/seo.jsp:215]: @lfbesada it should be camel case sitemap-include-child-pages, look for TypeSettingsProperties--

File: modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/display/context/LayoutsSEODisplayContext.java [hunk 2]
```diff
@@ -600,6 +601,22 @@ public List<SelectOption> getSitemapIncludeSelectOptions() {
...
 
+	public boolean isIncludeChildPagesInSitemap() {
+		Layout selLayout = getSelLayout();
+
+		UnicodeProperties layoutTypeSettingsUnicodeProperties =
+			selLayout.getTypeSettingsProperties();
+
+		if (GetterUtil.getBoolean(
+				layoutTypeSettingsUnicodeProperties.getProperty(
+					"sitemap-include-child-pages", Boolean.TRUE.toString()))) {
```
Comment (ealonso) [inline modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/display/context/LayoutsSEODisplayContext.java:612]: Camelcase

File: modules/apps/layout/layout-seo-web/src/main/resources/META-INF/resources/layout/screen/navigation/entries/seo.jsp (hunk 1 above)
Comment (lfbesada) [reply modules/apps/layout/layout-seo-web/src/main/resources/META-INF/resources/layout/screen/navigation/entries/seo.jsp:215]: @ealonso Thanks for the review :heart: 

You can find both patterns by looking for TypeSettingsProperties-- but for the SEO SiteMapSettings we are always using this pattern. 
The existing typeSettingsProperties in this JSP are:

-  TypeSettingsProperties--sitemap-include--
-  TypeSettingsProperties--sitemap-priority--
-  TypeSettingsProperties--sitemap-changefreq--

To preserve the pattern I choosed TypeSettingsProperties--sitemap-include-child-pages--

IMHO it is weird adding a property to this type setting properties using another pattern. 

What do you think?

Comment (ealonso) [reply modules/apps/layout/layout-seo-web/src/main/resources/META-INF/resources/layout/screen/navigation/entries/seo.jsp:215]: Ok, it make sense for this case 😞

**Comment by ealonso** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:report:2536680687

**Comment by lfbesada** [comment]
ci:forward:force

**Pull request closed without merging**